static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx

```diff
@@ -236,6 +242,58 @@ export const renderPrependColumns =
     ];
   };
 
+export const renderMetricsGridHeaderCell =
+  ({location, widget, tableData, organization, onHeaderClick}: Props) =>
+  (column: TableColumn<keyof TableDataRow>, _columnIndex: number): React.ReactNode => {
+    const tableMeta = tableData?.meta;
+    const align = fieldAlignment(column.name, column.type, tableMeta);
+    const widgetOrderBy = widget.queries[0].orderby;
+    const sort: Sort = {
+      kind: widgetOrderBy.startsWith('-') ? 'desc' : 'asc',
+      field: widgetOrderBy.startsWith('-') ? widgetOrderBy.slice(1) : widgetOrderBy,
+    };
+    const canSort = isAggregateField(column.name);
+    const titleText = column.name;
+
+    function generateSortLink(): LocationDescriptorObject | undefined {
+      const columnSort =
+        column.name === sort.field
+          ? {...sort, kind: sort.kind === 'desc' ? 'asc' : 'desc'}
+          : {kind: 'desc', field: column.name};
+
+      return {
+        ...location,
+        query: {
+          ...location.query,
+          [WidgetViewerQueryField.SORT]:
+            columnSort.kind === 'desc' ? `-${columnSort.field}` : columnSort.field,
+          [WidgetViewerQueryField.PAGE]: undefined,
+          [WidgetViewerQueryField.CURSOR]: undefined,
+        },
+      };
+    }
+
+    return (
+      <SortLink
+        align={align}
+        title={<StyledTooltip title={titleText}>{titleText}</StyledTooltip>}
+        direction={sort.field === column.name ? sort.kind : undefined}
+        canSort={canSort}
+        generateSortLink={generateSortLink}
+        onClick={() => {
+          onHeaderClick?.();
+          trackAdvancedAnalyticsEvent('dashboards_views.widget_viewer.sort', {
+            organization,
+            widget_type: widget.widgetType ?? WidgetType.METRICS,
```

Comment:
If this is specifically called for a metrics header, should we just directly pass in `WidgetType.METRICS` and not worry about the `??`?
```suggestion
            widget_type: WidgetType.METRICS,
```